addon/components/uni-datepicker.js

```diff
@@ -16,13 +16,25 @@ export default Component.extend({
   dateFormat: 'MMMM YYYY',
   showDaysAround: true,
   disabledDates: [],
+  messages: [],
 
   isFirstMonth: computed('center', function() {
     return this.get('center').isSame(this.get('minDate'), 'month') ? 'disabled' : '';
   }),
   isLastMonth: computed('center', function() {
     return this.get('center').isSame(this.get('maxDate'), 'month') ? 'disabled' : '';
   }),
+  selectedMessage: computed('messages', function() {
+    let messagesArray = this.get('messages');
+
+    return messagesArray.sort(function(a, b) {
+      if (a.type === b.type) {
+        return 0;
+      }
+
+      return a.type === 'information' ? 1 : -1;
+    })[0];
```

Comment:
Do you want this to return undefined when the array is empty? I'd separate the sort from the arr[0] access too.